src/compiler/checker.ts

```diff
@@ -18312,7 +18313,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         return constraint && (isGenericObjectType(constraint) || isGenericIndexType(constraint)) ? cloneTypeParameter(p) : p;
     }
 
-    function isSimpleTupleType(node: TypeNode) {
+    function isSimpleTupleType(node: TypeNode): node is TupleTypeNode {

```

Comment:
This guard seems unsafe; it doesn't prove the negative

Reply:
I reverted this. I forgot that the negative case should also be considered here since it might have an unwanted effect on the CFA :v